api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java

```diff
@@ -1151,12 +1151,52 @@ public Appointment bookAppointment(Appointment appointment,
 		}
 
 		appointment.setStatus(AppointmentStatus.SCHEDULED);
-		return Context.getService(AppointmentService.class).saveAppointment(
+		Appointment savedAppnt = Context.getService(AppointmentService.class).saveAppointment(
 				appointment);
+		addNewStatusToHistory(savedAppnt);
+		return savedAppnt;
+	}
+
+	@Override
+	public AppointmentStatusHistory getPreviousAppointmentStatus(Appointment appointment) {
+		return appointmentStatusHistoryDAO.getPreviousAppointmentStatus(appointment);
+	}
+
+	@Override
+	@Transactional
+	public List<AppointmentStatusHistory> getAppointmentStatusHistories(Appointment appointment) {
+		return getAppointmentStatusHistoryDAO().getAppointmentStatusHistories(appointment);
+	}
+
+	@Override
+	public void addNewStatusToHistory(Appointment appointment) {
```

Comment:
Can we call this "updateAppointmentStatusHistory" and add a comment in AppointmentService explaining how it differs from "changeAppointmentStatus"... (if I'm understanding it correctly, while "changeAppointmentStatus" takes in an appointment and a status to update to, while the idea of this method is to check if the current status of the appointment matches the most recent status history and then update if necessary?)